libs/vkd3d/resource.c

```diff
@@ -3891,8 +3944,10 @@ void d3d12_descriptor_heap_cleanup(struct d3d12_descriptor_heap *descriptor_heap
     const struct vkd3d_vk_device_procs *vk_procs = &descriptor_heap->device->vk_procs;
     const struct d3d12_device *device = descriptor_heap->device;
 
-    VK_CALL(vkDestroyDescriptorPool(device->vk_device,
-            descriptor_heap->vk_descriptor_pool, NULL));
+    VK_CALL(vkDestroyBuffer(device->vk_device, descriptor_heap->uav_counters.vk_buffer, NULL));
+    VK_CALL(vkFreeMemory(device->vk_device, descriptor_heap->uav_counters.vk_memory, NULL));
```

Comment:
Do we need to unmap buffers before freeing it in Vulkan?

Reply:
No, `vkFreeMemory` says the following:
> If a memory object is mapped at the time it is freed, it is implicitly unmapped.